src/sync/chase_lev.rs

```diff
@@ -245,19 +262,25 @@ impl<T> Deque<T> {
     // to be read after we flag this buffer for reclamation. All stealers,
     // however, have their own epoch pinned during this time so the buffer will
     // just naturally be free'd once all concurrent stealers have exited.
+    //
+    // This method may only be called from the workers, as it modifies the bottom
+    // counter.
     unsafe fn swap_buffer<'a>(&self,
                               b: isize,
                               old: Shared<'a, Buffer<T>>,
                               buf: Buffer<T>,
                               guard: &'a epoch::Guard)
                               -> Shared<'a, Buffer<T>> {
+        // replace the old buffer with the new.
         let newbuf = Owned::new(buf);
         let newbuf = self.array.store_and_ref(newbuf, SeqCst, &guard);
+        // add the new size to the bottom and top counters as a way of signalling
+        // to current stealers that the buffer has been resized.
         let ss = newbuf.size() as isize;
-        self.bottom.store(b + ss, SeqCst);
+        self.bottom.store(b + ss, Relaxed);
         let t = self.top.load(SeqCst);
         if self.top.compare_and_swap(t, t + ss, SeqCst) != t {
-            self.bottom.store(b, SeqCst);
+            self.bottom.store(b, Relaxed);
```

Comment:
Specifically, this is the kind of thing that might not fly on ARM. See also at line 174 where we reload the bottom counter after swapping. Does ARM guarantee that (on the same thread, with relaxed atomics) we will read back the value we just wrote? It seems to me that it's a safe guarantee using non-atomic instructions, but maybe the relaxed semantics bypass the memory model's usual guarantees for sequential code?